src/TRestAxionXrayWindow.cxx

```diff
@@ -0,0 +1,318 @@
+/// \param name The name of the specific metadata. It will be used to find the
+/// corresponding TRestGeant4Metadata section inside the RML.
+///
+TRestAxionXrayWindow::TRestAxionXrayWindow(const char* cfgFileName, string name)
+    : TRestMetadata(cfgFileName) {
+    debug << "Entering TRestAxionXrayWindow constructor( cfgFileName, name )" << endl;
+
+    Initialize();
+
+    LoadConfigFromFile(fConfigFileName, name);
+}
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestAxionXrayWindow::~TRestAxionXrayWindow() {}
+
+///////////////////////////////////////////////
+/// \brief Initialization of TRestAxionXrayWindow members. It removes all gases.
+///
+void TRestAxionXrayWindow::Initialize() {
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    fEnergy.clear();
+    fTransmission.clear();
+}
+
+///////////////////////////////////////////////
+/// \brief It reads the data files from the corresponding material that
+/// needs to be found in the axiolib database with .sol extension.
+/// Usually placed under `data/axion/transmission/`
+///
+void TRestAxionXrayWindow::ReadMaterial() {
+    std::string materialFileName = SearchFile(fMaterial + ".sol");
+
+    debug << "TRestAxionXrayWindow::ReadMaterial. Reading file : " << materialFileName << endl;
+
+    if (!TRestTools::fileExists(materialFileName)) {
+        ferr << "TRestAxionXrayWindow::ReadMaterial( )" << endl;
+        ferr << "Material file not found : " << materialFileName << endl;
+        exit(1);
+    }
+
+    FILE* fin = fopen(materialFileName.c_str(), "rt");
+
+    fEnergy.clear();
+    fTransmission.clear();
+    double en, value;
+    while (fscanf(fin, "%lf\t%lf\n", &en, &value) != EOF) {
+        debug << "Energy : " << en << "eV -- Abs : " << value << endl;
+
+        fEnergy.push_back(en / 1000.);
+        fTransmission.push_back(TMath::Power(value, fThickness * units("um")));
+    }
+
+    debug << "Items read : " << fEnergy.size() << endl;
+
+    fclose(fin);
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the window transmission probability for the given energy (in keV)
+/// and window position, using energy linear interpolation.
+///
+/// For the case of patterned window (stripped or grid), it will return 1 if the strip is
+/// not hitted.
+///
+Double_t TRestAxionXrayWindow::GetTransmission(Double_t energy, Double_t x, Double_t y) {
+    if (fEnergy.size() == 0) ReadMaterial();
+
+    if ((x - fCenter.X()) * (x - fCenter.X()) + (y - fCenter.Y()) * (y - fCenter.Y()) > fRadius * fRadius)
+        return 0;
+
+    if (!HitsPattern(x, y)) return 1.;
+
+    Double_t energyIndex = GetEnergyIndex(energy);
+
+    // Transmission
+    double y2 = fTransmission[energyIndex + 1];
+    double y1 = fTransmission[energyIndex];
+
+    // Normalized field
+    double x2 = fEnergy[energyIndex + 1];
+    double x1 = fEnergy[energyIndex];
```

Comment:
`GetEnergyIndex` returns `-1` for energies outside the range. That leads to hard to debug out of bounds access errors for bad inputs in the usage here.

Reply:
Fixed at d30f63f8